### PR TITLE
Persistent ToggleVignette and Improved Dupe Savestates

### PR DESCRIPTION
**Pull request by pseudorandomhk**
- `ToggleVignette` now persists between rooms
- Fixed extra black scene borders and camera jank after loading dupe savestates created right after setup (i.e. while still in the stag station), compared to setting up the dupe normally

### CODE REVIEW

File: Source/SaveStates/SaveState.cs [hunk 1]
```diff
@@ -148,19 +157,7 @@ public void LoadStateFromFile(int paramSlot)
...
                     {
-                        data.saveStateIdentifier = tmpData.saveStateIdentifier;
-                        data.cameraLockArea = tmpData.cameraLockArea;
-                        data.savedPd = tmpData.savedPd;
-                        data.savedSd = tmpData.savedSd;
-                        data.savePos = tmpData.savePos;
-                        data.saveScene = tmpData.saveScene;
-                        data.lockArea = tmpData.lockArea;
-                        data.isKinematized = tmpData.isKinematized;
-                        if (tmpData.loadedScenes is not null)
```
Comment (TheMulhima) [inline Source/SaveStates/SaveState.cs:159]: This check if important. It helps older made save state files load with new versions. You should also add a check for the new array you added to the data

File: Source/SaveStates/SaveState.cs [hunk 2]
```diff
@@ -275,6 +281,7 @@ private IEnumerator LoadStateCoro(bool loadDuped)
             typeof(HeroController)
                 .GetMethod("FinishedEnteringScene", BindingFlags.NonPublic | BindingFlags.Instance)?
                 .Invoke(HeroController.instance, new object[] {true, false});
+            ReflectionHelper.CallMethod(GameManager.instance, "UpdateUIStateFromGameState", Array.Empty<object>());
```
Comment (TheMulhima) [inline Source/SaveStates/SaveState.cs:284]: You can leave the last parameter blank if it doesn't need any parameters so like `ReflectionHelper.CallMethod(GameManager.instance, "UpdateUIStateFromGameState")`

File: Source/SaveStates/SaveState.cs (hunk 1 above)
Comment (pseudorandomhk) [reply Source/SaveStates/SaveState.cs:159]: Check is already done in the ctor (c.f. L55), this is just duplicate code

Comment (TheMulhima) [reply Source/SaveStates/SaveState.cs:159]: ah ok then nvm

File: Source/SaveStates/SceneWatcher.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                SceneData data = scenes.FirstOrDefault(d => d.id == id);
+                scenesWithManager.Remove(self.gameObject.scene);
+                
+                if (data != null)
+                    data.activeSceneWhenLoaded = GameManager.instance.sceneName;
+            };
+            USceneManager.sceneUnloaded += s => scenes.RemoveAt(scenes.FindIndex(d => d.name == s.name));
+        }
+        
+        public class SceneData
```
Comment (flibber-hk) [inline Source/SaveStates/SceneWatcher.cs:55]: Perhaps this class should not be called SceneData, to confuse with the SceneData class in HK

**Comment by TheMulhima** [comment]
Test comment

File: Source/SaveStates/SceneWatcher.cs (hunk 3 above)
Comment (TheMulhima) [reply Source/SaveStates/SceneWatcher.cs:55]: Loaded Scene Data?

Comment (flibber-hk) [reply Source/SaveStates/SceneWatcher.cs:55]: I think I like info more than data

Comment (pseudorandomhk) [reply Source/SaveStates/SceneWatcher.cs:55]: Sure, can commit tomorrow

**Pull request closed without merging**